### ISSUE TITLE
Product clone without matrix options: uncheck 'Options Matrix'

### ISSUE THREAD

**Issue by briansandall** [cubecart/v6#1156]
When cloning a product with the settings 'Product options' enabled and 'Product options matrix' disabled, any options that are included in the matrix of the original product are also checked as being included in the matrix of the current product, resulting in blank matrix entries being generated for all combinations.

What I would expect instead is the product options to all be copied but NOT marked as part of the matrix so that one may add, remove, or edit the options before creating a new matrix or even not creating a matrix at all.

Immediately unchecking all of the 'Options matrix' boxes for all options on a newly cloned product and then saving, for example, results in a bunch of unused matrix options appearing in the database (typically identifiable by the NULL value for each entry's cached_name).

**Comment by briansandall** [comment]
There are so many times I want to copy a product and all its options, remove or add one or two options, and then and only then create the matrix combinations.

What I'd do before resolving this issue is to copy a product, modify options as needed, delete all entries in CubeCart_option_matrix for the new product, alter the table to reset the auto-increment index, and then re-save the product to recreate valid option matrix entries.

I just got back to adding more products after a several month hiatus and immediately ran into this issue, and I finally decided enough was enough.
